arrivals/src/main/scala/com/pagerduty/arrivals/impl/headerauth/AuthHeaderDirectives.scala

```diff
@@ -0,0 +1,32 @@
+package com.pagerduty.arrivals.impl.headerauth
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.server.Directive0
+import com.pagerduty.arrivals.api.headerauth.HeaderAuthConfig
+import akka.http.scaladsl.server.Directives._
+import com.pagerduty.akka.http.support.RequestMetadata
+
+object AuthHeaderDirectives {
+
+  def addAuthHeader(
+      authConfig: HeaderAuthConfig
+    )(optAuthData: Option[authConfig.AuthData]
+    )(implicit reqMetadata: RequestMetadata
+    ): Directive0 = {
+    mapRequest { req =>
+      optAuthData match {
+        case Some(authData) => addAuthenticationHeader(authConfig)(req, authData)
+        case None           => req
```

Comment:
we passthrough the request if it fails auth?

Reply:
Yep, always have. This is a conscious choice - the API gateway should not need to know what paths require auth, and which don't. It's the responsibility of the upstream service to know that and check for the auth header as appropriate.